### ISSUE TITLE
Field :section_id is not a valid ID - Question

### ISSUE THREAD

**Issue by iloy1004** [gurock/trcli#111]
Hi, I'm having issue when trying to upload test results from XML to testRail.

The below is the command I used
`trcli -y -c "trcli-config.yml" parse_junit -f "reports/junit-report.xml"`

and this is the config file.
```
host: https://XXXX.testrail.io/
project: SMOKE TEST
username: XXX@XXXX.com
password: XXXX
title: Cypress Automated Test
file: reports/junit-report.xml
suite_id: 170
auto_creation_response: true
```

The below is what I've got, and I have no idea what's section id and why it's a problem because it was working well before.   Does anyone have any idea?  I would appreciate it if anyone could help me. 

```
trcli -y -c "trcli-config.yml" parse_junit -f "reports/junit-report.xml"
Checking project. Done.
Found test cases not matching any TestRail case (count: 1)
Adding missing test cases to the suite.
Adding test cases: 0/1
Error during add_case. Trying to cancel scheduled tasks.

Aborting: add_case. Trying to cancel scheduled tasks.
Adding test cases: 0/1
Field :section_id is not a valid ID.
```

**Comment by MattKiazyk** [comment]
I've come to realize that adding a `-v` onto the call gives you a verbose log that tells you what the actual issue is or at least leads you in the right direction.

**Comment by qmilangowin** [comment]
I'm getting this too. If I run each test case individually then I can upload but if I run the whole test-suite, I'm getting the same error as OP

**Comment by qmilangowin** [comment]
I realized what happened. The original tests were in a directory I was using WIP (for work in progress) and I uploaded the first batch from there.
After I moved them to my normal directory and there I get the error that the OP does. If I move them back to the WIP directory they will upload. The response body in the new directory still references the old location

**Comment by iloy1004** [comment]
> I've come to realize that adding a `-v` onto the call gives you a verbose log that tells you what the actual issue is or at least leads you in the right direction.

Thanks Matt,
I've got this details with -v as your suggestion.  do you happen to have any idea why api returns 400?  URL also seems incorrect as it ended "/add_case/None".  

`****
Found test cases not matching any TestRail case (count: 1)
Adding missing test cases to the suite.
Adding test cases: 0/1
**** API Call
method: POST
url: https://tactio.testrail.io/index.php?/api/v2/add_case/None
payload: {'title': 'Dashboard Dashboard Changing groups', 'custom_automation_id': 'Dashboard Changing groups.Dashboard Dashboard Changing groups'}
response status code: 400
response body: {'error': 'Field :section_id is not a valid ID.'}`

**Comment by d-rede** [comment]
Hello @iloy1004 

Thanks for bringing this to our attention. Can you provide a sample of the report with which we can reproduce the issue?

**Comment by JonathansDCI** [comment]
I also have this issue @d-rede. This is my first time working with testRailCLI and I was working my way to get the automated creation of test cases creating a proof of concept. I had our Testrail Admin create a new fake Project that has the automation_id as a string on test cases. There are no test cases, runs, etc. on this project.
[Test-Results.zip](https://github.com/gurock/trcli/files/10796834/Test-Results.zip)

**Comment by Testinator-X** [comment]
> I also have this issue @d-rede. This is my first time working with testRailCLI and I was working my way to get the automated creation of test cases creating a proof of concept. I had our Testrail Admin create a new fake Project that has the automation_id as a string on test cases. There are no test cases, runs, etc. on this project. [Test-Results.zip](https://github.com/gurock/trcli/files/10796834/Test-Results.zip)
> 
> **** API Call method: POST url: https://xxxxxxxx.testrail.net/index.php?/api/v2/add_case/0 payload: {'title': 'Test2', 'custom_automation_id': 'TestProject1.SmokeTests.Test2'} response status code: 400 response body: {'error': 'Field :section_id is not a valid section.'}
> 
> Adding test cases: 0/2 Field :section_id is not a valid section.

I‘ve checked your import file

`<testsuite name="ddm-api-beta.dll" tests="2" skipped="0" failures="1" errors="0" time="1.2098879999999999" timestamp="2023-02-21T18:57:23" hostname="JONATHANS-LT" id="0" package="ddm-api-beta.dll">`

The id you provide is `id="0"` but the id should match a section id you have in your project. I‘m quite sure there is no section with an ID of 0. change the ID to a section that exists in your project and it should work.

I think it’s quite the same problem as described here https://github.com/gurock/trcli/issues/71

**Comment by d-rede** [comment]
Hey everyone! 

Using the `id` attributes coming from the XML report was in fact undesired behavior and a fix was applied in `v1.4.3`. The `id` attributes are no longer being considered for sections and suites. 

**Note:** If you are using a multisuite project, you should use the `--suite-id` command line option to pass the id of the suite you want your test run created in.

**Comment by JonathansDCI** [comment]
My test results file was generated by https://github.com/spekt/junit.testlogger. I don't even know what a section id is.

**Comment by d-rede** [comment]
Hi there @JonathansDCI 

Can you install TestRail CLI version `1.4.3` and try importing your test results again? The issue with the IDs in the report should no longer occur.

**Comment by JonathansDCI** [comment]
IDK how to install a specific version, I don’t see any documentation except pip install trcli and that gives me version 1.4.1

**Comment by Testinator-X** [comment]
`C:\Path\To\Your\Python\Directory\python.exe -m pip install --upgrade trcli`

**Comment by JonathansDCI** [comment]
Works:
dotnet test --no-build --configuration Release --test-adapter-path:. --logger:"junit;LogFilePath=${REPORTS_PATH}/TestReport.xml"
trcli -y -c "trcli-config.yml" parse_junit -f "${REPORTS_PATH}/TestReport.xml"

**Comment by Testinator-X** [comment]
Lucky you! My problem from https://github.com/gurock/trcli/issues/71 has not been fixed, although the ticket has been closed. (Or I still don't know how to work around the problem via command line. )

**Comment by JonathansDCI** [comment]
Reading your bug #71 there is two slightly different issues, your error is complaining that you have a string value vs an int for id="foobar suite". My bug, which they fixed, is that https://github.com/spekt/junit.testlogger is outputting that ID="0" but I don't use that field at all (at least in a bare bones demo project). When I generate a junit file through the nunit console and transform it like the Sample Project, the id doesn't get generated in the junit file.

I suspect that although, there was a code change to ignore Id in the parsing, I suspect that there is some schema/transform/validation thingy still active before ID is even parsed, thus your id = string is still a problem,

**Comment by iloy1004** [comment]
Hello @JonathansDCI and @d-rede , thank you for your comments. I apologize for not checking the thread for a while. 

I resolved the issue by deleting some test cases in TestRail that I haven't developed scripts for in Cypress. Although I'm not entirely sure if this is the correct solution, the same command to send results to TestRail is now working fine after removing the extra test cases.

Previously, I added a few extra test cases to serve as guidelines when creating scripts, similar to blueprints. This resulted in the test suite having a greater number of test cases than what I actually implemented in the code.
